spec.yaml

```diff
@@ -218,6 +231,95 @@ definitions:
           (e.g. no use of special characters) providing more information
           on the cause of the error.
         example: RESOURCE_NOT_FOUND
+  Discovery:
+    type: object
+    required:
+      - enabled
+      - apiVersion
+      - endPoint
+      - resourceTypes
+    properties:
+      enabled:
+        type: boolean
+        description: Whether the OCM service is enabled at this endpoint
+      apiVersion:
+        type: string
+        description: The OCM API version this endpoint supports
+        example: 1.0.0
+      endPoint:
+        type: string
+        description: The URI of the OCM API available at this endpoint
+        example: https://my-cloud-storage.org/ocm
+      provider:
+        type: string
+        description: A friendly branding name of this endpoint
+        example: MyCloudStorage
+      resourceTypes:
+        type: array
+        description: A list of all supported resource types with their access protocols
+        items:
+          type: object
+          properties:
+            name:
+              type: string
+              description: |
+                A supported resource type (file, folder, calendar, contact, ...).
+                Implementations MUST support `file` at a minimum.
+              example: file
+            shareTypes:
+              type: array
+              description: |
+                The supported recipient share types.
+                Implementations MUST support `user` at a minimum.
+              items:
+                type: string
+                enum: ["user", "group", "federation"]
+              example: ["user"]
+            protocols:
+              type: object
+              description: |
+                The supported protocols to access shares at this endpoint.
+                Implementations MUST support `webdav` at a minimum.
+              anyOf:
+                - properties:
+                    webdav:
+                      type: string
+                      description: |
+                        The top-level WebDAV endpoint URI. In order to access a
+                        remote shared resource, implementations MAY use this
+                        URI as a prefix: e.g. if in a new share request's payload
+                        a `protocol.options.sharedSecret` property is defined
+                        (see singleProtocolLegacy example), it may need to be
+                        appended to this URI to access the resource.
+                      example: https://my-cloud-storage.org/remote.php/dav/ocm
```

Comment:
I guess we only want absolute URLs here, otherwise they would be relative to "something else" that is defined nowhere. Maybe we should explicitly say that a `top-level endpoint` is an absolute URL.